dali/pipeline/data/tensor_list.h

```diff
@@ -290,6 +291,24 @@ class DLL_PUBLIC TensorList : public Buffer<Backend> {
     tensor_views_.clear();
   }
 
+  /**
+   * @brief Swaps the content of two TensorLists
+   */
+  DLL_PUBLIC inline void Swap(TensorList<Backend> *other) {
+    std::swap(data_, other->data_);
+    std::swap(shape_, other->shape_);
+    std::swap(size_, other->size_);
+    std::swap(offsets_, other->offsets_);
+    std::swap(type_, other->type_);
+    std::swap(num_bytes_, other->num_bytes_);
+    std::swap(device_, other->device_);
+    std::swap(tensor_views_, other->tensor_views_);
+    std::swap(shares_data_, other->shares_data_);
+    std::swap(Buffer<Backend>::pinned_, other->pinned_);
+    std::swap(meta_, other->meta_);
+    std::swap(layout_, other->layout_);
```

Comment:
Wouldn't this handle it?
```suggestion
    if (&other != this)
      std::swap(*this, other);
```

Reply:
This function is gone now anyway.